time/time.go

```
@@ -47,3 +48,41 @@ func LoadLocation(name string) (*time.Location, error) {
 	locCache.cache[name] = loc
 	return loc, nil
 }
+
+// IncYear increments time t by count number of years
+func IncYear(t time.Time, count int) time.Time {
+	return t.AddDate(1*count, 0, 0)
+}
+
+// IncMonth increments time t by count number of months
+func IncMonth(t time.Time, count int) time.Time {
+	return t.AddDate(0, 1*count, 0)
+}
+
+// IncWeek increments time t by count number of weeks
+func IncWeek(t time.Time, count int) time.Time {
+	return t.AddDate(0, 0, (1*7)*count)
+}
+
+// IncDay increments time t by count number of days
+func IncDay(t time.Time, count int) time.Time {
+	return t.AddDate(0, 0, 1*count)
+}
+
+// IncHour increments time t by count number of hours
+func IncHour(t time.Time, count int) time.Time {
+	d, _ := time.ParseDuration(fmt.Sprintf("%vh", count))
```

Here's another way to accomplish the same thing without needing to swallow an error:
```
func IncHour(t time.Time, count int) time.Time {
  return t.Add(time.Hour * count)
}
```

Although with an implementation that simple I'm not sure we need a helper at all.

---

I agree with Mark. It may not be worth it if it's a one-liner like that. Also, there is a helper function in periodic that adds DST aware durations to a time: https://github.com/spothero/periodic/blob/main/periodic.go#L300-L308 I'm not sure if it makes most sense to use that for your use cases @noahwill, potentially wrap it with some more declarative functions as you have here, or if DST doesn't need to be considered.

---

Ah good catch - the week, minute, and second cases are the only ones that are annoying to do in line, the others can be one line but are included mostly for consistency

---

actually, now that I went to change it I see why the hours one was annoying as well - `time.Hour` is a duration and `count` is an int so you get a type error when trying to multiply. If it were `time.Hour * 1` (just that constant value) then there would be no error. Like I said, some of the ones that could be omitted since they are just one line are included for consistency in the library, so I think leaving this in is fine.

---

Oof yeah that's annoying, I guess you'd have to wrap it then. Still not so bad.
```
func IncHour(t time.Time, count int) time.Time {
  return t.Add(time.Hour * time.Duration(count))
}
```